packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ui/viewModel/InspectionsViewModel.kt

```diff
@@ -0,0 +1,28 @@
+package com.mongodb.jbplugin.ui.viewModel
+
+import com.intellij.openapi.components.Service
+import com.intellij.psi.PsiElement
+import com.mongodb.jbplugin.linting.QueryInsight
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.withContext
+
+@Service(Service.Level.PROJECT)
+class InspectionsViewModel {
+    val insights: MutableStateFlow<List<QueryInsight<PsiElement, *>>> = MutableStateFlow(emptyList())
```

Comment:
```suggestion
    private val mutableInsights: MutableStateFlow<List<QueryInsight<PsiElement, *>>> = MutableStateFlow(emptyList())
    val insights: StateFlow<List<QueryInsight<PsiElement, *>>>
      get() = mutableInsights
```
nit: because the insights won't be modified from outside, maybe we can expose the immutable flow?

Reply:
Changed 😄